sdks/java/core/src/main/java/org/apache/beam/sdk/util/gcsfs/GcsPath.java

```diff
@@ -601,6 +600,27 @@ public String toResourceName() {
     return sb.toString();
   }
 
+  private static final Pattern HAS_GLOB = Pattern.compile(".*(\\[|\\*|\\?).*");
+  public String getBrowseUrl() {
+    StringBuilder builder = new StringBuilder("https://console.cloud.google.com/storage/browser/");
```

Comment:
I've split off link URLs to a separate PR: https://github.com/apache/incubator-beam/pull/300